federation/pkg/dnsprovider/providers/coredns/coredns.go

```diff
@@ -48,11 +57,36 @@ func init() {
 	})
 }
 
+func newTransportForETCD2(certFile, keyFile, caFile string) (*http.Transport, error) {
```

Comment:
name should be more along lines of `etcdTLSTransport()`? I'm not sure the fact that it's the etcd v2 API specifically is relevant at the transport layer.